react-native-hub-app/src/checklist.tsx

```diff
@@ -0,0 +1,482 @@
+  }
+
+  async getUserThread(db: Client, ctx: Context): Promise<ThreadID> {
+    let idStr = await AsyncStorage.getItem(USER_THREAD_ID);
+    if (idStr) {
+      /**
+       * Temporary hack to get ThreadID working in RN
+       */
+      const temp: ThreadID = ThreadID.fromString(idStr);
+      // @ts-ignore
+      const id = ThreadID.fromBytes(Buffer.from(temp.buf));
+      return id;
+    } else {
+      const id: ThreadID = ThreadID.fromRandom();
+      await AsyncStorage.setItem(USER_THREAD_ID, id.toString());
+
+      /**
+       * Each new ThreadID requires a `newDB` call.
+       */
+      await db.newDB(id, ctx.withThread(id))
+
+      /** 
+       * We add our first Collection to the DB for Astronauts.
+       */
+      await db.newCollection(id, 'Astronaut', astronautSchema);
+      return id;
+    }
+  }
+
+  async runStep(stepNumber: number) {
+    const steps = this.state.steps;
+    const data = steps[stepNumber];
+    try {
+      let {ctx, db, threadId} = this.state;
+      switch (stepNumber) {
+        case 0: {
+          let updatedState = {};
+          /**
+           * Create a new user Identity
+           * 
+           * The identity pk will be cached in AsyncStorage.
+           * On the next session, the pk identity will be reused
+           */
+          const id = await this.generateIdentity();
+          
+          /**
+           * Context contains the token and session information
+           * 
+           * If possible, we'll reuse an existing session. 
+           * If it doesn't exist or is expired, we'll create a new one.
+           */
+          const existingCtx = await this.getContext();
+          if (existingCtx) {
+            db = new Client(existingCtx);
+            ctx = existingCtx;
+            data.message = 'Using existing Identity'
+          } else {
+            /** 
+             * Create a new Context (API_URL can be blank)
+             */
+            ctx = new Context(API_URL);
+
+            /**
+             * Create a new Signature from your key secret
+             * 
+             * note: this should be done offline
+             */
+            const sig = await createAPISig(USER_API_SECRET);
+
+            /**
+             * Add the new signature and key to the Context
+             */
+            ctx = ctx.withAPIKey(USER_API_KEY).withAPISig(sig);
+
+            /**
+             * Update our Database context
+             * 
+             * API calls will now include the credentials created above
+             */
+            db = new Client(ctx);
+
+            /**
+             * Generate an app user API token
+             * 
+             * The app user (defined by Identity) needs an API token
+             * The API will give you one based on ID plus Credentials
+             */
+            const token = await this.getUserToken(db, id);
+
+            /**
+             * Update our Context with the token
+             */
+            ctx = ctx.withToken(token);
```

Comment:
Note also that if you do end up calling `getToken` then `ctx.withToken` is called under the hood with the client's context.